### PR TITLE
Dat 2602 - add ponto linker for native apps to link in marker popup thumbnails

### PR DESCRIPTION
**Pull request by RafLeszczynski**

### CODE REVIEW

**Comment by RafLeszczynski** [comment]
@rogatty @nandy-andy @bkoval could you please take a look?

File: assets/scripts/im.pontoWikiaAppsBridge.js [hunk 1]
```diff
@@ -27,8 +26,9 @@ define('im.pontoWikiaAppsBridge', ['ponto', 'im.window'], function (ponto, w) {
 			var target = event.target;
 
 			if (
-				target.tagName === 'A' &&
-				target.className.indexOf(poiArticleLinkClass) !== -1
+				target.tagName === 'A' || target.tagName === 'IMG' &&
```
Comment (rogatty) [inline assets/scripts/im.pontoWikiaAppsBridge.js]: You probably want it to be `(target.tagName === 'A' || target.tagName === 'IMG') &&`

Comment (RafLeszczynski) [inline assets/scripts/im.pontoWikiaAppsBridge.js]: true :-) good catch :+1:

File: assets/scripts/im.pontoWikiaAppsBridge.js [hunk 2]
```diff
@@ -27,8 +26,9 @@ define('im.pontoWikiaAppsBridge', ['ponto', 'im.window'], function (ponto, w) {
...
+				target.className.indexOf(config.articleLinkClassName) !== -1 ||
+				target.parentNode.className.indexOf(config.articleLinkClassName) !== -1
```
Comment (rogatty) [inline assets/scripts/im.pontoWikiaAppsBridge.js]: The same here, `&&` has a higher precedence than `||`
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#Table

**Comment by rogatty** [comment]
LGTM

**Comment by nandy-andy** [comment]
LGTM :+1:

**Comment by rogatty** [comment]
Still LGTM

File: assets/scripts/im.pontoWikiaAppsBridge.js [hunk 3]
```diff
@@ -18,21 +17,39 @@ define('im.pontoWikiaAppsBridge', ['ponto', 'im.window'], function (ponto, w) {
 	}
 
 	/**
+	 * @desc checks if target is valid link to article
+	 * @param {Element} target - HTML node element
+	 * @returns {Element|Boolean} - if no valid article link element exist returns false
+	 */
+	function getArticleLinkEl(target) {
```
Comment (nandy-andy) [inline assets/scripts/im.pontoWikiaAppsBridge.js]: Unit test? :japanese_ogre:

**Comment by nandy-andy** [comment]
And still EiA for me :smile_cat: Consider adding a unit test, though :stuck_out_tongue_winking_eye:

File: assets/scripts/im.pontoWikiaAppsBridge.js [hunk 4]
```diff
@@ -18,21 +17,39 @@ define('im.pontoWikiaAppsBridge', ['ponto', 'im.window'], function (ponto, w) {
...
+
+		if (target.tagName === 'A' && target.className.indexOf(config.articleLinkClassName) !== -1) {
+			link = target;
+		} else if (
+			target.tagName === 'IMG' &&
+			target.parentNode.className.indexOf(config.articleLinkClassName) !== -1
+		) {
+			link = target.parentNode;
+		} else {
+			link = false;
```
Comment (rogatty) [inline assets/scripts/im.pontoWikiaAppsBridge.js]: Let's return null instead of false.

**Comment by RafLeszczynski** [comment]
@nandy-andy please review again with unit tests :-)

**Comment by nandy-andy** [comment]
Everything is awesome!

:sheep: it... :smile_cat:

**Comment by aniaMu** [comment]
Ping

**Pull request merged**